content/ManipulatingFiles/ManipulatingFiles.q.md

```diff
@@ -0,0 +1,1029 @@
+
+ - `$file`
+
+ - `$myFile`
+
+ - `$fileToRead`
+
+ - `$copiedFile`
+
+ - `$myCopiedFile`
+
+
+/// type=MS, answer=[2,4]
+
+Which of the following are file handling functions?
+
+ - `die()`
+
+ - `copy()`
+
+ - `copyFile()`
+
+ - `readfile()`
+
+ - `readFileContent()`
+
+
+/// type=MS, answer=[1,2,3,4]
+
+Which of the following are function calls?
+
+ - `readfile($fileToRead)`
+
+ - `copy($file, $copiedFile)`
+
+ - `readFileContent($myCopiedFile);`
+
+ - `copyFile($myFile, $myCopiedFile);`
+
+ - `function readFileContent($fileToRead) {... }`
+
+
+/// type=SS, answer=[3]
+
+Which of the following is a magic constant?
+
+ - `die()`
+
+ - `copy()`
+
+ - `__DIR__`
+
+ - `$myFile`
+
+ - `readFile()`
+
+
+/// type=SS, answer=[1]
+
+On line 4, what does the `copy($file, $copiedFile)` function call do?
+
+ - It copies the content of `$file` to `$copiedFile`.
+
+ - It copies the content of `$copiedFile` to `$file`.
+
+ - It accesses the content of `$file` in `$copiedFile`.
+
+ - It accesses the content of `$copiedFile` in `$file`.
+
+ - It removes the content of `$file` from `$copiedFile`.
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the `copyFile()` function?
+
+ - It opens the specified file as append-only.
+
+ - It creates the specified file if it does not exist.
+
+ - It reads the content of the specified file line by line.
+
+ - It copies the content of the specified file to another file.
+
+ - It is a user-defined function that requires two arguments.
+
+
+/// type=MS, answer=[1,4]
+
+On line 9, what does the `readfile($fileToRead)` function call do?
+
+ - It reads the specified `$fileToRead`.
+
+ - It deletes the specified `$fileToRead` from its directory.
+
+ - It creates the specified `$fileToRead` if it does not exist.
+
+ - It writes the specified `$fileToRead` to the output buffer.
+
+ - It copies the content of the specified `$fileToRead` to another file.
+
+
+/// type=MS, answer=[1,4,5]
+
+Which statements correctly describe the `readFileContent()` function?
+
+ - It reads the specified file.
+
+ - It removes the content of the specified file.
+
+ - It creates the specified file if it does not exist.
+
+ - It writes the specified file to the output buffer.
+
+ - It is a user-defined function that requires one argument.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+function renameFile($fileToRename, $newName)
+{
+   rename($fileToRename, $newName) or die("Unable to rename the file."); 
+}
+
+$file = __DIR__ . "/testDocument.txt";
+renameFile($file, "myRenamedDocument.txt");
+echo readlink(__DIR__ . "/myRenamedDocument.txt");
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+ - No output is displayed.
+
+ - It prints `Unable to rename the file.`.
+
+ - It prints `This is a test document content.`.
+
+ - It prints `This is the end of the test document content.`.
+
+ - It prints `C:\xampp\htdocs\PhpProjectFileHandling\myRenamedDocument.txt`.
```

Comment:
I think it will not be the same path/output when this runs in Codestop.

Reply:
Yes, this is known during content creation and noted already. I just temporarily provide this here until such time that we can test this in our reader when multiple files is ready. But, for now we will keep this as is. Thank you.

Reply:
Ok, this noted. Thanks.